resolwe_bio/processes/reads_processing/trimmomatic.yml

```diff
@@ -131,65 +151,111 @@
-      re-progress 0.7
-
-      mv out_reads_filtered.fastq.gz "${NAME}_filtered.fastq.gz"
-
-      echo "Postprocessing FastQC..."
-
-      # http://www.bioinformatics.babraham.ac.uk/projects/fastqc/
-      mkdir "fastqc" && fastqc "${NAME}_filtered.fastq.gz" --extract --outdir="fastqc" >&2
-      re-checkrc "Failed while processing with FastQC."
-      re-progress 0.9
-
-      mv "fastqc/${NAME}_filtered_fastqc" "fastqc/${NAME}_fastqc"
-      mv "fastqc/${NAME}_filtered_fastqc.zip" "${NAME}_fastqc.zip"
-
-      FASTQC_URL="{\"file\":\"fastqc/${NAME}_fastqc/fastqc_report.html\",\"refs\":[\"fastqc/${NAME}_fastqc\"]}"
-
-      re-progress 0.95
-
-      re-save-file-list fastq "${NAME}_filtered.fastq.gz"
-      re-save-file stats stats.txt
+      {% if
+        not reads_filtering.average_quality
+        and not reads_filtering.minlen
+        and not trim_bases.headcrop
+        and not trim_bases.crop
+        and not trim_bases.trailing
+        and not trim_bases.leading
+        and not maxinfo.target_length
+        and not  maxinfo.strictness
+        and not slidingwindow.window_size
+        and not slidingwindow.required_quality
+        and not illuminaclip.adapters
+        and not illuminaclip.seed_mismatches
+        and not illuminaclip.simple_clip_threshold %}
+        re-warning "To process reads with Trimmomatic please define at least one parameter
+      (Iluminaclip and/or Window size ans/or Maxinfo and/or Leading and/or Trailing ond/or Crop and/or Headcrop and/or
+      Min length and/or Average quality)."
+
+        cp in_reads.fastq.gz "${NAME}.fastq.gz"
+        cp {{ reads.fastqc_archive.0.file }} "${NAME}_fastqc.zip"
+
+        mkdir "fastqc"
+        cp -r {{ reads.fastqc_url.0.refs.0}} "fastqc/${NAME}_fastqc"
+
+        FASTQC_URL="{\"file\":\"fastqc/${NAME}_fastqc/fastqc_report.html\",\"refs\":[\"fastqc/${NAME}_fastqc\"]}"
+
+        re-save-file-list fastq "${NAME}.fastq.gz"
+        re-save-file-list fastqc_archive "${NAME}_fastqc.zip"
+        re-save-list fastqc_url "${FASTQC_URL}"
+
+      {% else %}
+
+        echo "Removing adapters and filtering data..."
+
+        {% if illuminaclip.adapters and not illuminaclip.seed_mismatches and not illuminaclip.simple_clip_threshold %}
+          re-warning "Illuminaclip was not preformed because only adapter file was provided, please specify also
+      Seed mismatches and Simple clip threshold parameters."
+        {% else %}
+          echo "Preparing adapter file..."
+          ADAPTERS=`basename {{illuminaclip.adapters.fasta.file}} .fasta.gz`
+          gzip -cd {{ illuminaclip.adapters.fasta.file }} > "${ADAPTERS}.fasta"
+          re-progress 0.1
+        {% endif %}
+
+        {% if (slidingwindow.window_size and not slidingwindow.required_quality)
+          or (slidingwindow.required_quality and not slidingwindow.window_size) %}
+          re-warning "Slidingwindow was not preformed because only Window size or only Required quality were defined.
```

Comment:
Use labels consistently, i.e. put _Window size_ and _Required quality_ in quotes in description of inputs.